projects/igniteui-angular/src/lib/grids/grid/grid.pagination.spec.ts

```diff
@@ -1,557 +1,435 @@
+            GridFunctions.verifyGridPager(fix, 3, '1', '2\xA0of\xA02', [false, false, true, true]);
+        });
 
-        // Add new rows on second page
-        grid.addRow({ ID: 2, Name: 'Test Name', JobTitle: 'Test Job Title' });
-        grid.addRow({ ID: 3, Name: 'Test Name', JobTitle: 'Test Job Title' });
-        grid.addRow({ ID: 4, Name: 'Test Name', JobTitle: 'Test Job Title' });
-        fix.detectChanges();
-        verifyGridPager(fix, 3, '1', '2\xA0of\xA03', [false, false, false, false]);
-        expect(grid.totalPages).toBe(3);
+        it('should not throw when initialized in a grid with % height', () => {
 
-        // Go to last page and delete the row
-        grid.nextPage();
-        fix.detectChanges();
-        grid.deleteRow(4);
-        fix.detectChanges();
-        verifyGridPager(fix, 3, '1', '2\xA0of\xA02', [false, false, true, true]);
-    });
+            grid.paging = true;
+            expect(() => {
+                fix.detectChanges();
+            }).not.toThrow();
+        });
+
+        it('"paginate" method should paginate correctly', () => {
+            GridFunctions.testPagingAPI(fix, grid, (pageIndex) => grid.paginate(pageIndex));
+        });
+
+        it('"page" property should paginate correctly', () => {
+            GridFunctions.testPagingAPI(fix, grid, (pageIndex) => grid.page = pageIndex);
+        });
+
+        it('should hide paginator when there is no data or all records are filtered out.', () => {
+
+            GridFunctions.verifyGridPager(fix, 3, '1', '1\xA0of\xA04', [true, true, false, false]);
+
+            // Filter out all records
+            grid.filter('ID', 1000, IgxNumberFilteringOperand.instance().condition('greaterThan'));
+            fix.detectChanges();
+
+            expect(GridFunctions.getGridPaginator(grid)).toBeNull();
+
+            grid.filter('ID', 1, IgxNumberFilteringOperand.instance().condition('greaterThan'));
+            fix.detectChanges();
+            expect(GridFunctions.getGridPaginator(grid)).not.toBeNull();
+
+            grid.data = null;
+            fix.detectChanges();
+            expect(GridFunctions.getGridPaginator(grid)).toBeNull();
 
-    it('should not throw when initialized in a grid with % height', () => {
-        const fix = TestBed.createComponent(GridIDNameJobTitleComponent);
-        const grid = fix.componentInstance.grid;
-        grid.paging = true;
-        expect(() => {
+            grid.data = fix.componentInstance.data;
             fix.detectChanges();
-        }).not.toThrow();
+            expect(GridFunctions.getGridPaginator(grid)).not.toBeNull();
+        });
     });
 
-    it('"paginate" method should paginate correctly', fakeAsync(() => {
-        const fix = TestBed.createComponent(PagingComponent);
-        fix.detectChanges();
-        const grid = fix.componentInstance.grid;
-        testPagingAPI(fix, grid, (pageIndex) => grid.paginate(pageIndex));
-    }));
+    it('change paging with button', () => {
```

Comment:
This test can be make without clicking on buttons and just direct calling of the methods and this is done in the test 'should paginate data API', so you can remove the test

Reply:
Done